src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/Handler/XamlSolutionProvider.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.LanguageServer.Handler
+{
+    [Export(typeof(XamlSolutionProvider)), Shared]
+    internal class XamlSolutionProvider : ILspSolutionProvider
+    {
+        private readonly XamlProjectService _projectService;
+        private readonly ILspSolutionProvider _solutionProvider;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public XamlSolutionProvider(ILspSolutionProvider lspSolutionProvider, XamlProjectService projectService)
+        {
+            _solutionProvider = lspSolutionProvider;
+            _projectService = projectService;
+        }
+
+        public Solution GetCurrentSolutionForMainWorkspace()
+        {
+            return _solutionProvider.GetCurrentSolutionForMainWorkspace();
+        }
+
+        public ImmutableArray<Document> GetDocuments(Uri documentUri)
+        {
+            if (documentUri.IsAbsoluteUri)
+            {
+                _projectService.TrackOpenDocument(documentUri.LocalPath);
```

Comment:
a little confused on this - why do we track the localpath instead of the absolute here, but pass the absolute to the solution provider?  It seems like for this to work they would both need to be the same?

Reply:
@dibarbet  [Uri.LocalPath](https://docs.microsoft.com/en-us/dotnet/api/system.uri.localpath?view=netframework-4.7.2) represents how the path is represented on the local OS, and that's what I need to find the document in the RDT (the AbsolutePath still has forward slashes in it).